File: Data Structures & Algorithms/decode-ways/submission-1.py
class Solution:
    
    def dfs(self, i, s):
        
        if i == len(s):
            return 1
        if s[i] == '0' :
            return 0
        
        total_count = self.dfs(i+1,s)
        if i+1 < len(s) :
            if s[i] == '1' or (s[i] == '2' and s[i+1] < '7') :
                total_count +=  self.dfs(i+2,s)
        return total_count

    def numDecodings(self, s: str) -> int:
        return self.dfs(0,s)
        

File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-1.cpp
class Solution {
public:
    
    int dfs(int index, vector<int>& cost){
        if (index > cost.size()){
            return 0;
        }
        return cost[index] + std::min(dfs(index+1,cost),dfs(index+2,cost));
    }

    int minCostClimbingStairs(vector<int>& cost) {
        return std::min(dfs(0,cost),dfs(1,cost));
    }
};
